### ISSUE TITLE
No figure is shown in running py-faster-rcnn demo!

### ISSUE THREAD

**Issue by mayyphyokhaing** [rbgirshick/py-faster-rcnn#631]
I'm now running py-faster-rcnn demo in my Ubuntu 16.04 on Oracle VirtualBox in CPU mode.

I have this error '_tkinter.TclError: no display name and no $DISPLAY environment variable' when my backend in vi /usr/local/lib/python2.7/dist-packages/matplotlib/mpl-data/matplotlibrc
backend: TkAgg

So, I changed backend: Agg and the error is gone. But, the problem is that no figures are shown. 
Loaded network /home/ubuntu/py-faster-rcnn/data/faster_rcnn_models/ZF_faster_rcnn_final.caffemodel
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/000456.jpg
Detection took 13.975s for 300 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/000542.jpg
Detection took 3.413s for 135 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/001150.jpg
Detection took 3.979s for 231 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/001763.jpg
Detection took 3.768s for 200 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/004545.jpg
Detection took 4.362s for 300 object proposals

Please, somebody help me with this.

Best Regard.

**Comment by eugene123tw** [comment]
Maybe try to install tkinter ```sudo apt-get install python-tk ```?

**Comment by mayyphyokhaing** [comment]
I have already installed tkinter sudo apt-get install python-tk.
The figure is still not shown !
Have any idea, please??

**Comment by maximus009** [comment]
This could happen if the model is not detecting any object with confidence greater than CONF_THRESH.
Also, the size of the image matters, so try tweaking the scale values in lib/fast_rcnn/config.py

Hope that helps!

**Comment by mayyphyokhaing** [comment]
Thanks you all and I have now solved my problem.

The problem is that I was just only running py-faster-rcnn on Ubuntu built on the VirtualBox and my Ubuntu is just the command line with no desktop screen(which means no $DISPLAY environment). 
After I install my Ubuntu desktop via
$tasksel
SELECT Ubuntu desktop, 
I got the Ubuntu desktop screen on my VirtualBox and the figure is shown.
(P.S the backend select is set to TkAgg on  /usr/local/lib/python2.7/dist-packages/matplotlib/mpl-data/matplotlibrc)
